### PR TITLE
Bugfix/fixing tests

### PR DESCRIPTION
**Pull request by drewpts**

### CODE REVIEW

File: libproject/test/libprojecttest.cpp [hunk 1]
```diff
@@ -298,6 +280,9 @@ class TestRemoveSubprojects : public TestSkeleton {
     case_6_pathToMainFileWhichWillNotBeLoaded,
```
Comment (mkwork) [inline libproject/test/libprojecttest.cpp:280]: _#bug_
case_6_
case_7_
e.t.c
Are bad names - them don't give info about what is it.

File: libproject/src/libproject.cpp [hunk 2]
```diff
@@ -258,40 +258,42 @@ namespace Interface {
...
-                throw FileSetRuntimeError(FileSetRuntimeError::BrokenSubproject, "Trying to add broken subproject(s)");
-            checkSubprojectsStream.close();
-
-            if(jChangedContentOfProjectFile.count("subprojects") == 0)
-                jChangedContentOfProjectFile["subprojects"] = { };
-            jChangedContentOfProjectFile["subprojects"].push_back(relativePath);
+            candidates.push_back(relativePath);
         }
 
+        if (candidates.size() > 1)
```
Comment (mkwork) [inline libproject/src/libproject.cpp:286]: Too big nesting level. Is it really neccesary?

File: libproject/test/libprojecttest.cpp [hunk 3]
```diff
@@ -298,6 +280,9 @@ class TestRemoveSubprojects : public TestSkeleton {
...
     case_7_pathToMainFileWhereWillBeRemoveAndCountActions;
 
+    string case_0_contentBackup,
```
Comment (mkwork) [inline libproject/test/libprojecttest.cpp:283]: #style
We didn't have supplied code style, but in most of styles that are presented - it's forbidden to declare variables in comma separated way.

**Comment by mkwork** [comment]
I expected fix for bug with teardowning the tests which can be easily reviewed and merged to _develop_
but see massive refactoring and "bugfixing". Please don't put your wishes in task's implementation.

File: libproject/test/libprojecttest.cpp (hunk 1 above)
Comment (drewpts) [inline libproject/test/libprojecttest.cpp:280]: Fixed.

File: libproject/src/libproject.cpp (hunk 2 above)
Comment (drewpts) [inline libproject/src/libproject.cpp:286]: Fixed by using std::set.

File: libproject/test/libprojecttest.cpp (hunk 3 above)
Comment (drewpts) [inline libproject/test/libprojecttest.cpp:283]: Fixed.

**Pull request closed without merging**